swap/src/protocol/alice/event_loop.rs

```diff
@@ -161,8 +162,26 @@ where
                                     continue;
                                 }
                             }
-
-                            let state0 = match State0::new(btc, xmr, self.env_config, self.bitcoin_wallet.as_ref(), &mut OsRng).await {
+                            // TODO: This should be cleaned up.
+                            let tx_redeem_fee = self.bitcoin_wallet
+                                .estimate_fee(bitcoin::TxRedeem::weight())
+                                .await.unwrap();
```

Comment:
Why are you blowing up the whole event-loop here if this fails? As far as I can see, this can definitely be hit by IO errors happening in the underlying electrum-client!